### PR TITLE
splits out gossip CrdsData out of crds_value.rs

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Reorganizing this code in preparation of upcoming changes to `CrdsValue`  (de)serialization.

#### Summary of Changes
split out gossip `CrdsData` out of `crds_value.rs`

### CODE REVIEW

File: gossip/src/crds_data.rs [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+            CrdsData::Version(version) => version.from,
+            CrdsData::NodeInstance(node) => node.from,
+            CrdsData::DuplicateShred(_, shred) => shred.from,
+            CrdsData::SnapshotHashes(hash) => hash.from,
+            CrdsData::ContactInfo(node) => *node.pubkey(),
+            CrdsData::RestartLastVotedForkSlots(slots) => slots.from,
+            CrdsData::RestartHeaviestFork(fork) => fork.from,
+        }
+    }
+}
```
Comment (gregcusack) [inline gossip/src/crds_data.rs:176]: unrelated to this exact PR, but why are `pubkey` and `wallclock` private in `ContactInfo` and `LegacyContactInfo` but public in the other `CrdsData` types

File: gossip/src/crds_data.rs [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+                MAX_EPOCH_SLOTS,
+                EpochSlots::new(keypair.pubkey(), timestamp()),
+            ),
+            &keypair,
+        );
+        assert_eq!(item.sanitize(), Err(SanitizeError::ValueOutOfBounds));
+    }
+
+    #[test]
+    fn test_node_instance_crds_lable() {
```
Comment (gregcusack) [inline gossip/src/crds_data.rs]:
```suggestion
    fn test_node_instance_crds_label() {
```

File: gossip/src/crds.rs [hunk 3]
```diff
@@ -199,17 +200,22 @@ fn overrides(value: &CrdsValue, other: &VersionedCrdsValue) -> bool {
     // Contact-infos and node instances are special cased so that if there are
     // two running instances of the same node, the more recent start is
     // propagated through gossip regardless of wallclocks.
-    if let CrdsData::NodeInstance(value) = value.data() {
-        if let Some(out) = value.overrides(&other.value) {
-            return out;
+    match value.data() {
```
Comment (gregcusack) [inline gossip/src/crds.rs:203]: should we just pass `data: &CrdsData` into this function instead?

Comment (behzadnouri) [reply gossip/src/crds.rs:203]: I think if we pass `CrdsData`, it kind of obfuscates `{ContactInfo,NodeInstance}::overrides` api.
Might be more self-explanatory if the api uses a more specific type;

File: gossip/src/crds_data.rs (hunk 1 above)
Comment (behzadnouri) [reply gossip/src/crds_data.rs:176]: A lot of these variants are defined in this same module.
So even if they were private, this code could still access them directly.

**Review by gregcusack — APPROVED** [review]
lgtm!

**Pull request merged**